opensrp-app/src/main/java/org/smartregister/repository/EventClientRepository.java

```diff
@@ -1203,6 +1255,144 @@ public JSONObject getEventsByBaseEntityIdAndEventType(String baseEntityId, Strin
         return list;
     }
 
+    /**
+     * Fetches {@link Event}s whose rowid > #lastRowId up to the #limit provided. Each jsonObject contains the
+     * default properties as the one fetched from the DB with an additional property that holds the {@code syncStatus}
+     * and {@code rowid} which are used for peer-to-peer sync.
+     *
+     * @param lastRowId
+     * @return JsonData which contains a {@link JSONArray} and the maximum row id in the array
+     *          of {@link Event}s returned. This enables this method to be called again for the consequent batches
+     */
+    @Nullable
+    public JsonData getEvents(long lastRowId, int limit) {
+        JsonData jsonData = null;
+        JSONArray jsonArray = new JSONArray();
+        long maxRowId = 0;
+
+        String query = "SELECT "
+                + event_column.json
+                + ","
+                + event_column.syncStatus
+                + ","
+                + ROWID
+                + " FROM "
+                + Table.event.name()
+                + " WHERE "
+                + ROWID
+                + " > ? "
+                + " ORDER BY " + ROWID + " ASC LIMIT ?";
+        Cursor cursor = null;
+
+        try {
+            cursor = getWritableDatabase().rawQuery(query, new Object[]{lastRowId, limit});
+
+            while (cursor.moveToNext()) {
+                String jsonEventStr = (cursor.getString(0));
+                long rowId = cursor.getLong(2);
+                String syncStatus = cursor.getString(1);
+
+                if (StringUtils.isBlank(jsonEventStr)
+                        || jsonEventStr.equals("{}")) { // Skip blank/empty json string
+                    continue;
```

Comment:
should you check the below before continuing

```
 if (rowId > maxRowId) {
                    maxRowId = rowId;
                }

````

Reply:
The `maxRowId` is used to define the `highestRecordId` inside current `JsonData`, it serves it's purpose. It's not u